runbot/static/src/js/fields/fields.js

```diff
@@ -135,6 +136,29 @@ registry.category("fields").add("char_frontend_url", {
     component: FieldCharFrontendUrl,
 });
 
+
+// Pull Request URL Widget
+class PullRequestUrlField extends UrlField {
+    static template = xml`
+    <a class="o_field_widget o_form_uri" t-on-click.stop="" t-att-href="formattedHref" t-esc="props.record.data[props.name].replace('https://github.com/odoo','').replace('/pull','') || ''" target="_blank"/>
+    `;
+    static components = { UrlField }
+    setup() {
+        if (!this.props.readonly) {
+            throw new Error("This widget works only on readonly fields");
+        }
+    }
+}
```

Comment:
I would consider this instead:
 - Use a regex to get the parts of the url we need.
 - Use the same format as used on github <user>/<repo>#issueId.
 - Remove custom template -> prefer using the existing field and compute the `text` props ourselves.

I removed the readonly constraint but we could definitely keep it.

```suggestion
// Pull Request URL Widget
const pullRequestRegex = /\/([a-zA-Z-_]+\/[a-zA-Z-_]+)\/pull\/(\d+)/;
class PullRequestUrlField extends UrlField {
    static template = xml`
        <UrlField t-props="fieldProps"/>
    `;
    static components = { UrlField }

    get fieldProps() {
        const props = {...this.props};
        const parts = pullRequestRegex.exec(this.props.record.data[props.name])
        if (parts) {
            props.text = `${parts[1]}#${parts[2]}`;
        }
        return props
    }
}
```